react-and-redux-style-guide/README.md

```diff
@@ -0,0 +1,469 @@
+# RingCentral React and Redux Style Guide
+
+- [Introduction](#introduction)
+- [Code structure](#code-structure)
+- [Naming](#naming)
+- [Good practices](#good-practices)
+
+# Introduction
+
+This is a set of recommended patterns and best practices for writing applications using React+Redux.
+
+# Code structure
+
+## # Put all Redux codes in `redux` directory inside your module 
+> Reason: this helps to keep module isolated
+
+## # Keep reducers clean by moving logic to helpers
+> Reason: a lot of logic inside the reducer complicates an understanding of data changes produced by the action
+```typescript
+// BAD
+export const someReducer: Reducer<SomeStore> = (state: SomeStore = initialState, action: ActionTypes) => {
+    switch (action.type) {
+            case UPDATE: {
+                let oldItems = state.items;
+                let someData = getSomeDraftData(oldItems).map(() => {
+                    ...
+                });       
+                ...
+                let items = someData.reduce((acc, item) => {
+                    ...
+                }, {});
+                return {
+                    ...state,
+                    items,
+                };
+            }
+            default: {
+                return state;
+            }
+    }
+};
+```
+```typescript
+// GOOD
+export const someReducer: Reducer<SomeStore, ActionTypes> = (state = initialState, action) => {
+    switch (action.type) {
+            case UPDATE: {
+                return {
+                    ...state,
+                    items: calculateItemsUpdates(state.items),
+                };
+            }
+            default: {
+                return state;
+            }
+    }
+};
+``` 
+ 
+## # Put Redux connection logic to Containers
+> Reason: this makes the connection between a React component and Redux storage more transparent and isolated.
+```typescript
+// EXAMPLE
+// /someModule/containers/FilterContainer.tsx
+import {Filter} from '../components/settings/Filter';
+import {itemsSelector} from '../redux/selectors';
+import {updateFilter} from '../redux/actions';
+
+const mapStateToProps = (state) => ({
+    items: itemsSelector(state),
+});
+
+const mapDispatchToProps = {
+    onChange: updateFilter,
+};
+
+export const FilterContainer: React.FC<{}> = connect(
+    mapStateToProps,
+    mapDispatchToProps,
+)(Filter);
+```
+
+## # All connected components should be defined in a separate files
```

Comment:
Why? Provide reasoning. In general it does not make sense to split ALL components like that, it violates high cohesion principle.

Reply:
It makes code reading easier. There are a lot of cases when both container and connected component grows big enough to live in a separate file. So placing them according to this rule helps to avoid additional refactoring in the future.

Reply:
I strongly recommend NOT to use separate connected and dumb components. Best way is to have both in the same file, export both. Connected component should just pass props, that's it. Otherwise it quickly becomes a nightmare to support fat connected components and fat dumb components.